### ISSUE TITLE
[Bug] Training with 16k mono wav files

### ISSUE THREAD

**Issue by exosome12** [deezer/spleeter#586]
Hi,

I am trying to train spleeter with 16k mono wav files but I have to convert them to stereo and leave "sample_rate" at "44100" and "n_channels" at "2" in the config json to start the actual training, otherwise it just creates the cache file and stands there forever. 

This would be fine, but cache file size matters which is acceptable with the desired settings. I have read after the issue and did trial/errors but could not find answer/solution. 

It is happening on:
- Windows 10 
- nvidia gtx 1050
- tried this repo+poetry and also the simple pip install

I have also tried on google collab. Same results. 

Any help would be greatly appreciated!

**Comment by romi1502** [comment]
Hi @exosome12,
I think that your problem comes from [this function](https://github.com/deezer/spleeter/blob/4ef593dc94d820fb019236cd799d03c858acd7fc/spleeter/dataset.py#L241) that filters out from the dataset all samples that don't have the proper size. This filtering is done to avoid having incorrectly shaped samples, for instance mono samples in the middle of a stereo dataset, or samples with a shorter time duration than the expected size duration, that would results in an error when trying to batch samples of different shapes together.
In your case (mono 16kHz), this function filters out the whole dataset and that's why it stands out for ever.
For some reasons the number of channels of samples was hard coded to `2`, instead of taking into account the `n_channels` parameter, so all your mono samples are filtered out.

I'll have a look if we can fix this easily. In the meantime, you can:
- check if it works with 16k stereo files.
- use a custom version of spleeter by replacing the hard coded `2` value with `1` for the number of channels in `dataset.py`, but there may be some other places where stereo is hard coded.

**Comment by romi1502** [comment]
I just fixed this in this [PR](https://github.com/deezer/spleeter/pull/588) that was just merged to master and pushed on pip (version 2.2.0). You should be able to train a model with mono files with this last version.

By the way, be aware that you don't necessarily need to reduce the sample rate to reduce cache size, you can just reduce the `F` option which is the number of FFT bins until which the spectrogram will be processed (and cached). For instance, for 44.1kHz signals with a frame_length of 4096, taking F=768 will only process frequencies up to 8.2kHz, which is roughly the same frequency range that you would process with a 16kHz sample rate. Note that `F` has to be a multiple of 64 for the provided U-net model.

**Comment by xavriley** [comment]
I was also trying to train with mono files. I've pulled the latest master which includes those changes but I'm seeing a different error now:

```
Traceback (most recent call last):
  File "/Users/xavriley/opt/anaconda3/envs/spleeter/lib/python3.8/site-packages/tensorflow/python/client/session.py", line 1365, in _do_call
    return fn(*args)
  File "/Users/xavriley/opt/anaconda3/envs/spleeter/lib/python3.8/site-packages/tensorflow/python/client/session.py", line 1349, in _run_fn
    return self._call_tf_sessionrun(options, feed_dict, fetch_list,
  File "/Users/xavriley/opt/anaconda3/envs/spleeter/lib/python3.8/site-packages/tensorflow/python/client/session.py", line 1441, in _call_tf_sessionrun
    return tf_session.TF_SessionRun_wrapper(self._session, options, feed_dict,
tensorflow.python.framework.errors_impl.InvalidArgumentError: input depth must be evenly divisible by filter depth: 1 vs 2
	 [[{{node conv2d/BiasAdd}}]]
```

For now I've converted the files to stereo with `sox input.wav output.wav channels 2` and it's training. I just wanted to let others know that mono might not be working as expected yet.
